.test-infra/tools/stale_cleaner.py

```diff
@@ -0,0 +1,208 @@
+        blob = bucket.blob(f"stale_cleaner/{self.resource_type}-deleted")
+        blob.upload_from_filename("/tmp/deleted_resources.bin")
+
+    # Save the new project state
+    def update(self, bucket_name) -> None:
+        # Get working resource list
+        now_resources_dict = self.get_now_resources()
+        working_resource_dict = self.get_working_resources(bucket_name)
+        deleted_resource_dict = self.get_deleted_resources(bucket_name)
+
+        working_resources_to_delete = []
+
+        # Go through the working resource list comparing the working list with the new list
+        for resource_name, resource_obj in working_resource_dict.items():
+            # If no longer exists, add it to the deleted dictionary
+            if resource_name not in now_resources_dict:
+                deleted_resource_dict[resource_name] = resource_obj
+                working_resources_to_delete.append(resource_name)
+            # If it exists, update the last update date
+            else:
+                working_resource_dict[resource_name].update()
+                now_resources_dict.pop(resource_name)
+
+        # Delete the resources that are no longer alive
+        for resource_name in working_resources_to_delete:
+            working_resource_dict.pop(resource_name)
+
+        # Add all the new resources to the working dictionary
+        for resource_name, resource_obj in now_resources_dict.items():
+            working_resource_dict[resource_name] = resource_obj
+
+        # Save the working and deleted resources
+        self.set_working_resources(working_resource_dict, bucket_name)
+        self.set_deleted_resources(deleted_resource_dict, bucket_name)
+
+# Create a new PubSubCleaner object
+class PubSubCleaner(StaleCleaner):
+    def __init__(self, project_id) -> None:
+        super().__init__(project_id)
+        self.client = pubsub_v1.PublisherClient()
+        self.resource_type = "pubsub"
+
+    def get_now_resources(self) -> defaultdict:
+        d = defaultdict(GoogleCloudResource)
+        for topic in self.client.list_topics(request={"project": self.project_path}):
+            d[topic.name] = GoogleCloudResource(topic.name, self.resource_type)
+        return d
+
+# Local testing
+if __name__ == "__main__":
+    cleaner = PubSubCleaner("apache-beam-testing")
+    cleaner.update("apache-beam-testing-pabloem")
+
+    print("Now resources")
+    d = cleaner.get_now_resources()
+    for k, v in d.items():
+        print(f"{k} -> {v.time_alive()}")
+
+    print("Working resources")
+    d = cleaner.get_working_resources("apache-beam-testing-pabloem")
+    for k, v in d.items():
+        print(f"{k} -> {v.time_alive()}")
+
+    print("Deleted resources")
+    d = cleaner.get_deleted_resources("apache-beam-testing-pabloem")
+    for k, v in d.items():
+        print(f"{k} -> {v.time_alive()}")
+
+    print("Old resources")
+    d = cleaner.get_old_resources("apache-beam-testing-pabloem", 60)
```

Comment:
let's make the time threshold also a constructor parameter (it will be usually the same for the same kind of resource (e.g. maybe 6 days for GCS bucket, but 1 day or less for pubsub topic, etc).